benchmarks/src/test/java/org/apache/druid/benchmark/query/SqlWindowFnBenchmark.java

```diff
@@ -0,0 +1,433 @@
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Benchmark that tests various SQL queries.
+ */
+@State(Scope.Benchmark)
+@Fork(value = 1)
+@Warmup(iterations = 3)
+@Measurement(iterations = 5)
+public class SqlWindowFnBenchmark
+{
+  static {
+    NullHandling.initializeForTests();
+  }
+
+  private static final Logger log = new Logger(SqlWindowFnBenchmark.class);
+
+  private static final String STORAGE_MMAP = "mmap";
+  private static final String STORAGE_FRAME_ROW = "frame-row";
+  private static final String STORAGE_FRAME_COLUMNAR = "frame-columnar";
+
+  @Param({"2000000"})
+  private int rowsPerSegment;
+
+  @Param({
+      "auto"
+  })
+  private String schema;
+
+  // Can be STORAGE_MMAP, STORAGE_FRAME_ROW, or STORAGE_FRAME_COLUMNAR
+  @Param({STORAGE_MMAP})
+  private String storageType;
+
+  private SqlEngine engine;
+
+  @Nullable
+  private PlannerFactory plannerFactory;
+  private final Closer closer = Closer.create();
+
+  private static final DruidProcessingConfig PROCESSING_CONFIG = new DruidProcessingConfig() {
+
+    @Override
+    public int getNumMergeBuffers()
+    {
+      return 3;
+    }
+  };
+
+  @Setup(Level.Trial)
+  public void setup()
+  {
+    final GeneratorSchemaInfo schemaInfo = GeneratorBasicSchemas.SCHEMA_MAP.get("basic");
+    final DataSegment dataSegment = schemaInfo.makeSegmentDescriptor("foo");
+    final SegmentGenerator segmentGenerator = closer.register(new SegmentGenerator());
+
+    log.info("Starting benchmark setup using cacheDir[%s], rows[%,d].", segmentGenerator.getCacheDir(), rowsPerSegment);
+    final QueryableIndex index;
+    if ("auto".equals(schema)) {
+      List<DimensionSchema> columnSchemas = schemaInfo.getDimensionsSpec()
+                                                      .getDimensions()
+                                                      .stream()
+                                                      .map(x -> new AutoTypeColumnSchema(x.getName(), null))
+                                                      .collect(Collectors.toList());
+      index = segmentGenerator.generate(
+          dataSegment,
+          schemaInfo,
+          DimensionsSpec.builder().setDimensions(columnSchemas).build(),
+          TransformSpec.NONE,
+          IndexSpec.builder().withStringDictionaryEncoding(getStringEncodingStrategy()).build(),
+          Granularities.NONE,
+          rowsPerSegment
+      );
+    } else {
+      index = segmentGenerator.generate(dataSegment, schemaInfo, Granularities.NONE, rowsPerSegment);
+    }
+
+    final Pair<PlannerFactory, SqlEngine> sqlSystem = createSqlSystem(
+        ImmutableMap.of(dataSegment, index),
+        Collections.emptyMap(),
+        null,
+        closer
+    );
+
+    plannerFactory = sqlSystem.lhs;
+    engine = sqlSystem.rhs;
```

Comment:
I guess this was copied from somewhere?
`createSqlSystem` is a static method which returns a pair which gets assigned to class fields this seems pretty complicated...

we can have it like this for now - but it would be better to extract not relevant common pieces to a shared class or something...